web/app/Controller/PasswordChangeController.php

```diff
@@ -28,23 +28,39 @@ public function Index(): void
     }
     $password_old = $_POST['password'];
     $password_new = $_POST['password2'];
+    $password_new_2 = $_POST['password3'];
+    if ($password_new_2 !== $password_new) {
+        redirect(url('PasswordChange', Null, ['Status'=>'ERROR_E'])->getAbsoluteUrl());
+    }
+
 
     if (password_verify($password_old, $password_hash) == true) {
         $password_new_hash = password_hash($password_new, PASSWORD_DEFAULT);
+        if ($password_new == $password_old) {
+            redirect(url('PasswordChange', Null, ['Status'=>'ERROR_EE'])->getAbsoluteUrl());
+        }
+        if ($password_new_2 == $password_old) {
+            redirect(url('PasswordChange', Null, ['Status'=>'ERROR_EE'])->getAbsoluteUrl());
+        }
+
         $sql = "UPDATE person SET password = '$password_new_hash' WHERE username = '$username'";
         $result = $connection->getConnection()->query($sql);
         // Konfiguration
-        $mail = new mailpassword();
-        $mail->GetEmail();
-
+        $mail = new PHPMailer();
+        $mail->isSMTP(); // SMTP verwenden
+        $mail->Host = 'smtp.office365.com'; // SMTP-Server für Microsoft 365
+        $mail->SMTPAuth = true; // SMTP-Authentifizierung aktivieren
+        $mail->Username = 'maximilian.schwarz@igs-edigheim.de'; // SMTP-Benutzername (deine Microsoft 365 E-Mail-Adresse)
```

Comment:
das steht alles an andere stelle schonmal. sollte in eine mail klasse, damit der code nicht wiederholt wird und bugs/erweiterungen an verschiedenen stellen gemacht werden müssen